include/dmlc/logging.h

```diff
@@ -319,14 +319,14 @@ inline std::string Demangle(char const *msg_str) {
   return string(msg_str);
 }
 
-inline std::string StackTrace() {
+inline std::string StackTrace(
+    const int stack_size = DMLC_LOG_STACK_TRACE_SIZE) {
```

Comment:
Any reason to make the parameter const? I haven't seen any examples where non-pointer parameters are made const.

Reply:
Also let's make the parameter unsigned, so that negative inputs won't be allowed.

Reply:
Here `const` is just an habit.